spec/jobs/create_derivatives_job_spec.rb

```diff
@@ -28,5 +29,37 @@
     it "does not error with a non-existent file_set_id" do
       expect { described_class.perform_now("blabla") }.not_to raise_error
     end
+
+    context "when an ImageMagick error is raised" do
+      let(:derivatives_service) { instance_double(Valkyrie::Derivatives::DerivativeService) }
+
+      before do
+        allow(Rails.logger).to receive(:error)
+        allow(derivatives_service).to receive(:create_derivatives).and_raise(MiniMagick::Error)
+        allow(derivatives_service).to receive(:cleanup_derivatives)
+        allow(Valkyrie::Derivatives::DerivativeService).to receive(:for).with(id: file_set.id).and_return(derivatives_service)
+        allow(described_class).to receive(:perform_later)
+      end
+
+      it "logs and error and retries the job" do
+        described_class.perform_now(file_set.id)
+        expect(Rails.logger).to have_received(:error).with("Failed to create the derivatives for #{file_set.id}: MiniMagick::Error")
+
+        expect(derivatives_service).to have_received(:create_derivatives).once
+        expect(described_class).to have_received(:perform_later)
+      end
+    end
+  end
+
+  context "with ImageMagick uninstalled on your MacBook" do
+    let(:file) { fixture_file_upload("files/abstract.tiff", "image/tiff") }
+    let(:scanned_resource) { FactoryBot.create_for_repository(:scanned_resource, files: [file]) }
+    let(:file_set) { scanned_resource.decorate.file_sets.first }
+    let(:adapter) { Valkyrie::MetadataAdapter.find(:indexing_persister) }
+    let(:query_service) { adapter.query_service }
+
+    it "propagates an ImageMagick error", run_real_derivatives: true do
+      expect { described_class.perform_now(file_set.id.to_s) }.to raise_error(MiniMagick::Invalid)
```

Comment:
This does successfully propagate when I uninstall ImageMagick within my local development environment, thus I am uncertain as to how this would be failing silently if it the error is being raised by `MiniMagick`.

Reply:
Yeah.. that makes me think we haven't found the right trigger for the silent fail.